### DISCUSSION TITLE
Question about included resources.

### DISCUSSION THREAD

**Discussion by pasoulier** [django-json-api/django-rest-framework-json-api#948] [Q&A]
When I'm querying a url that use several level of serializers (eg: /knight) I need to specify the included resources in the url otherwise I'm getting an error message `This endpoint does not support the include parameter for path story`. I'm only having this problem with the current version (4.2.0). I'm new using DRF and JSONAPI.

I'm made a simple app that you can see below.

Thanks for your help.

my models.py
```python
from django.db import models

# Create your models here.
class Knight(models.Model):
    name = models.CharField(max_length=50)
    nickname = models.CharField(max_length=50)

class Story(models.Model):
    name = models.CharField(max_length=50)

class Quest(models.Model):
    name = models.CharField(max_length=50)
    knights = models.ManyToManyField(Knight, related_name='quests')
    story = models.ForeignKey(Story, on_delete=models.CASCADE, related_name='quest')

```

urls.py

```python
from django.urls import path, include
from rest_framework import routers
from rest_framework_json_api import serializers
from rest_framework_json_api.views import ModelViewSet
from rest_framework_json_api.relations import ResourceRelatedField

from .models import Knight, Story, Quest
from . import views

class StorySerializer(serializers.ModelSerializer):
    quest = ResourceRelatedField(queryset=Quest.objects, many=True)

    included_serializers = {
        'quest': 'knight.urls.QuestSerializer',
    }

    class Meta:
        model = Story
        fields = ('id', 'name', 'quest')

    class JSONAPIMeta:
        included_resources = ['quest']

class QuestSerializer(serializers.ModelSerializer):
    story = ResourceRelatedField(queryset=Story.objects)

    included_serializers = {
        'story': 'knight.urls.StorySerializer',
    }

    class Meta:
        model = Quest
        fields = ('id', 'name', 'story',)

    class JSONAPIMeta:
        included_resources = ['story']

class KnightSerializer(serializers.ModelSerializer):
    quests = ResourceRelatedField(
        queryset=Quest.objects,
        many=True,
    )

    included_serializers = {
        'quests': 'knight.urls.QuestSerializer',
    }

    class Meta:
        model = Knight
        fields = ('id', 'name', 'nickname', 'quests',)

    class JSONAPIMeta:
        included_resources = ['quests']

class KnightViewSet(ModelViewSet):
    queryset = Knight.objects.all()
    serializer_class = KnightSerializer

class QuestViewSet(ModelViewSet):
    queryset = Quest.objects.all()
    serializer_class = QuestSerializer

class StoryViewSet(ModelViewSet):
    queryset = Story.objects.all()
    serializer_class = StorySerializer

router = routers.DefaultRouter()
router.register(r'/knights', KnightViewSet)
router.register(r'/quests', QuestViewSet)
router.register(r'/stories', StoryViewSet)


urlpatterns = [
    path('', include(router.urls)),
]
```

**Comment by sliverc** [comment]
The change #900 in version 4.2.0 most likely causes this issue. The question is though whether there is an error in your setup (before #900 there was no validation done on included resources) or whether there is a regression in DJA itself.

When I look at your sample it looks fine to me. Is this a made up example or actual code you use?

[At](https://github.com/django-json-api/django-rest-framework-json-api/blob/master/example/serializers.py#L180) there is an example which we use in our test suite which passes.

Are you able to reproduce this issue with in the DJA example app? Or if it is easier you could also create a small example app yourself which reproduces the issue and post the repo here. Quick and dirty is fine as long as there is code which I can run on my machine easily to better discover the issue.

**Reply by pasoulier** [reply]
It's prettry close from the code that I'm using but not with "real" data.

Here for the [repository](https://github.com/awfulbadger/poc_knight)

Thanks again for your time

**Reply by sliverc** [reply]
Thanks for the example repository. It is indeed a regression in DJA. I have create a bug report at #949
